### PR TITLE
Event Caching in local storage

### PR DESCRIPTION
**Pull request by Durbatuluk1701**
Some initial caching attempts

### CODE REVIEW

File: src/src/components/EventsStore/index.ts [hunk 1]
```diff
@@ -101,6 +101,18 @@ export function useEvents(
...
+export function useCachedEvents(): EventsStore | undefined {
+  const cachedEventsStr = window.localStorage.getItem('events');
+  if (!cachedEventsStr) return undefined;
+  const cachedEvents = JSON.parse(cachedEventsStr) as EventsStore;
+  if (!cachedEvents) return undefined;
+  return cachedEvents;
+}
+
+export function setCachedEvents(events: EventsStore): void {
+  window.localStorage.setItem('events', JSON.stringify(events));
```
Comment (maxxxxxdlp) [inline src/src/components/EventsStore/index.ts:117]: This is not correct. This is accessing Google Chrome's local storage.
Extensions have their own storage that they should use - https://developer.chrome.com/docs/extensions/reference/storage/

The API is quite similar though so should be easy to change

File: src/src/components/Core/App.tsx [hunk 2]
```diff
@@ -34,18 +39,23 @@ export function App(): JSX.Element | null {
...
-  const eventsStore = React.useRef<EventsStore>({});
+  const eventsStore = React.useRef<EventsStore>(useCachedEvents() ?? {});
   const durations = useEvents(
     eventsStore,
     // Don't fetch until the overlay is opened
     isOpen ? currentView?.firstDay : undefined,
     currentView?.lastDay
   );
 
+  React.useEffect(() => {
```
[the remark this answers is not in the document]
Comment (maxxxxxdlp) [reply src/src/components/Core/App.tsx]: I implemented an example solution using browser's native events API - https://github.com/maxxxxxdlp/calendar-plus/pull/80/commits/6fe2841b3efa603374f40237fdc3a69fead9246c

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]

File: src/src/components/Core/App.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Durbatuluk1701) [reply src/src/components/Core/App.tsx]: So do we think this is resolved. Also thank you for the help

File: src/src/components/EventsStore/index.ts (hunk 1 above)
Comment (Durbatuluk1701) [reply src/src/components/EventsStore/index.ts]: I believe this should be good now

**Review by maxxxxxdlp — APPROVED** [review]

**Pull request merged**